src/Chapter4.hs

```diff
@@ -488,7 +493,18 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+concatenate :: List a -> List a -> List a
+concatenate x Empty = x
+concatenate Empty x = x
```

Comment:
Nice catch!

You won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤